code/segmentedtp/pubsub.q

```diff
@@ -0,0 +1,106 @@
+// Pub/sub utilities for segmented tp process
+// Functionality for clients to subscribe to all tables or a subset
+// Includes option for subsrcibe to apply filters to received data
+
+/load schema from params, default to "sym.q"
+.proc.loadf[(src:$[`schemafile in key .proc.params;raze .proc.params`schemafile;"sym"]),".q"];
+
+\d .stpps
+
+t:tables`.
+
+// Handles to publish all data
+subrequestall:t!(count t)#()
+
+// Handles and conditions to publish filtered data
+subrequestfiltered:([tabname:`$()]handle:`int$();filts:`$();columns:`$())
+
+msgcount:t!(count t)#0
+
+// Function to send end of period messages to subscribers
+// Assumes that .u.endp has been defined on the client side
+endp:{
+  (neg union/[subrequestall])@\:(`.u.endp;x;y);
+  (neg union/[subrequestfiltered[;`handle]])@\:(`.u.endp;x;y);
```

Comment:
Is it possble that the same handle is in both of these lists so would get the same message twice?